### PR TITLE
Micro lesson proposal

### PR DESCRIPTION
**Pull request by BillMills**
Here's my first draft of the micro-lessons proposal. Please note, I parameterized a couple of things that should be discussed further by the mentorship SC:

 - recommend inviting new instructors to teach a micro lesson immediately following instructor training, and follow-up again 6 months after instructor training for instructors who haven't taught yet.
 - solidified the recommendation we discussed that a micro-lesson can provide an alternative route to badging
 - proposed a couple of metrics to assess how the program is doing.

Also note, most of the mentorship SC members aren't listed in `subcommittees.md` - plz everyone add yourselves!

### CODE REVIEW

File: subcommittees/mentoring/microLessonProposal.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+## Micro-Lessons: Building Positive First Experiences in New Software Carpentry Instructors
+
+### Overview
+
+It has been observed [reference?] that a large fraction [number?] of people completing Software Carpentry (SWC) Instructor Training haven't yet taught a workshop, possibly due in part to the logistical challenges of first teaching with an experienced instructor. Furthermore, a great deal of the anonymous minute-card feedback collected after live Instructor Training events has reported misgivings felt by students on their level of familiarity with SWC material, and a lack of confidence that they were ready to teach an entire unit at a full workshop. Given the high demand for SWC workshops, the Mentorship Committee suspects the disconnect between Instructor Training and getting in front of a classroom may also be one of instructor confidence and experience, and may bridged by a smaller, guided teaching experience that serves as a stepping stone to a full workshop, and helps relax the logistical challenges of co-teaching with an experienced instructor.
```
Comment (jiffyclub) [inline subcommittees/mentoring/microLessonProposal.md]: I asked @gvwilson about this and he indicated that the bottleneck for new instructors is not their hesitancy, but that we have more instructors coming on board than we have workshops in which to schedule them.

**Comment by wking** [comment]
On Mon, Mar 23, 2015 at 11:42:39AM -0700, Bill Mills wrote:
> Here's my first draft of the micro-lessons proposal.

This looks good to me.  An hour (and a half) is a reasonable chunk of
time to cover something cool, especially if you know your audience
well (e.g. your lab group).  Do we expect to maintain (example?)
lesson-template-based content for these micro lessons, or are the
instructors supposed to trim down an existing lesson or start from
scratch to get something that fits their audience more tightly?  On
the trimming route, assuming the time estimates for the git-novice
pages are pretty accurate, you should be able to fit:

  01-setup.md:minutes: 10
  02-create.md:minutes: 10
  03-changes.md:minutes: 20
  04-history.md:minutes: 15

into an hour, and:

  06-collab.md:minutes: 60
  07-conflict.md:minutes: 30

into an hour and a half.  If your audience was more experienced than
the usual SWC novice student, you could probably squeeze those down
enough to give space for the usual milling about that happens at the
beginning and end of any meeting ;).

You could fit a slightly-fast version of the whole shell lesson into
an hour and a half:

  shell-novice $ grep '^minutes:' *.md
  00-intro.md:minutes: 5
  01-filedir.md:minutes: 15
  02-create.md:minutes: 15
  03-pipefilter.md:minutes: 15
  04-loop.md:minutes: 15
  05-script.md:minutes: 15
  06-find.md:minutes: 15

but dropping 'script' and 'find' would give you more breathing room.
Although I'm not sure how much change you could effect with such a
quick introduction to the shell.  The real usefulness for the shell
comes from scripting and learning the large set of existing utilities
(like find), and skipping that material makes the shell much less
appealing.  On the other hand, an hour is certainly enough time to
introduce a few of the more advanced ideas [1] to make existing shell
users more efficient.

[1]: https://github.com/swcarpentry/shell-extras/issues/1

File: subcommittees/mentoring/microLessonProposal.md (hunk 1 above)
Comment (wking) [inline subcommittees/mentoring/microLessonProposal.md]: On Mon, Mar 23, 2015 at 12:50:14PM -0700, Matt Davis wrote:
> I asked @gvwilson about this and he indicated that the bottleneck
> for new instructors is not their hesitancy, but that we have more
> instructors coming on board than we have workshops in which to
> schedule them.

So when do the “Train the Admin” classes start?

**Comment by jiffyclub** [comment]
Pinging @swcarpentry/lessons for comments and general awareness because this idea would involve new lesson development and maintenance.

**Comment by jiffyclub** [comment]
Shorter lessons are something I want to explore, and not just in the context of new instructors. The availability of shorter lessons could make it possible for everyone to teach more often and in new contexts, such as lab groups, local clubs, and meetups. Our two-day lessons are one optimization of a challenging set of criteria: instructor time, student time, host time, and content to cover. But I think it's time to explore how much work it would be to develop shorter lessons, how effective they could be, and whether people are interested in delivering them. This is one of the primary things I plan to work on with the @swcarpentry/lessons committee.

**Comment by BillMills** [comment]
@jiffyclub I'd like to understand a bit more what @gvwilson means when he says the bottleneck is simply the number of workshops offered; in a lot of places, all we have to do to up the number of workshops is announce them - they fill as fast as we can book them. Is the problem that these instructors are in smaller towns or institutions that don't support the same demand for events? Greg, would appreciate your thoughts here - what is the geographic distribution of the instructors we're trying to reach?

Also Matt re: smaller lessons, I heartily agree! We're assembling a collection of lessons and pointers to lessons over at the [Study Groups Project](https://github.com/mozillascience/studyGroupLessons) at MSL; my attitude there has been to let the community drive on what lessons they want to see, by assembling and promoting whatever is getting taught in their current study groups. I would love to work together with you on this - maybe we should discuss that offlist.

@wking, you've got the right idea about teaching subsets of the existing lessons - but I would cut even harder than you are presently. No need to rush through lots of sections or cram stuff in; that's what the full workshops are for :) In my 'To-Do' section at the bottom of the proposal, I mention that we should tease out subsets of existing lessons, just like you're doing there, to make this process smooth and easy for new instructors.

**Comment by jiffyclub** [comment]
I will let @gvwilson expound when he has time, but I expect we are not dealing with a demand problem. Our new instructors likely have demand at their institutions, but in order to run a workshop we also need a local organizer/host willing to provide a room, travel for at least one experienced instructor, and our admin fee. I think hosts are our bottleneck.

**Comment by BillMills** [comment]
@jiffyclub ahh ok, so it's partly a central-organization problem, if the SCF is administrating the workshop scheduling - what about encouraging local instructors to roll their own? Gets our numbers up without blowing an admin gasket (and yes, I realize we want the admin fee - but we aren't getting it in these cases anyway if that service is 100% saturated).

On the rooms and hosts side, I think that's actually the demand / geography problem I was wondering about; in Vancouver, there's always someone at UBC or SFU happy to book us a room, email their department when reg opens, and have us show up. If the demand is there, there's usually a faculty member who will help us out with all that stuff. Not true in other towns?

I agree that getting an experienced lead instructor to lead the workshop is hard, because they're either there or they're not - no amount of local industriousness can solve that one. I suspect the bottleneck is either finding them, or, as the mentorship committee is speculating, the need for a clearer and simpler path between Instructor Training and that first workshop.

**Comment by wking** [comment]
On Mon, Mar 23, 2015 at 03:23:51PM -0700, Bill Mills wrote:
> We're assembling a collection of lessons and pointers to lessons
> over at the Study Groups Project at MSL; my attitude there has been
> to let the community drive on what lessons they want to see, by
> assembling and promoting whatever is getting taught in their current
> study groups.

+1 for community driven and centrally linked.  Looking at the
studyGroupLessons repository, I see one inline lesson on awk [1] and
some links to other groups/lessons [2].  I'd suggest keeping the
lessons themselves out of the study-group repository, since
maintaining a lesson on awk and keeping track of study groups around
the world teaching scientists our tooling seem like two different
tasks.  And linking to lessons by title seems better than linking to
groups.  At least, there should be a way to answer “what are the most
popular hour-long lessons talking about awk?” (or whatever).

> I would love to work together with you on this - maybe we should
> discuss that offlist.

Maybe not in this particular issue, but I'd love to be a part of this
discussion happening in a public forum somewhere.  Maybe another
swcarpentry/board issue?  Or a studyGroupLessons issue?

[1]: https://github.com/mozillascience/studyGroupLessons/tree/8a10e65cc241698ce511d69d559f0c95ddae3027/awk
[2]: https://github.com/mozillascience/studyGroupLessons/tree/8a10e65cc241698ce511d69d559f0c95ddae3027#check-out-our-friends

**Comment by jiffyclub** [comment]
@BillMills, I do not mean that so many hosts are asking for workshops that we don't have the admins to schedule them all. I mean that we don't have enough hosts getting in touch to schedule workshops.

**Comment by BillMills** [comment]
@jiffyclub ah I see, okay then - but so, workshops are really popular when they happen, but not enough people are asking for them? Sounds like an outreach problem. As @karinlag rightly pointed out, the Micro-Lesson program could be a great opportunity to bring little 'taster samples' of SWC places, to build interest and get those potential hosts getting in touch.

**Comment by jiffyclub** [comment]
No doubt potential hosts have any number of reasons they don't ask for a workshop, but certainly more SWC events in more places is a great way to make sure "I've never heard of Software Carpentry" isn't among those reasons.

**Comment by BillMills** [comment]
cool - just rewrote the intro to be more about outreach and less about instructor reticence, let me know.

**Comment by daisieh** [comment]
My two cents: this is a great idea. I did an ad-hoc version of this when I taught Git and Github to the non-devs at Dryad, and that was the route that made me feel more confident about teaching at SWC and also led to Greg giving me my instructor certification. However, I definitely found the breadth and size of the SWC git materials to be overwhelming, which led to me rewriting big swaths for my own benefit. More structured microlessons would probably be extremely beneficial for others doing this. I think the recent change to the git-novice lesson that breaks it up into smaller parts goes a long way towards achieving the microlesson strategy.

**Comment by ScienceScribe** [comment]
For students with a basic grasp of R already (i.e. they know how to load, view, and subset data sets), maybe a Micro-Lesson on dplyr could provide dramatic benefits.

This article makes a convincing case for focusing on dplyr: http://www.statsblogs.com/2014/02/10/how-dplyr-replaced-my-most-common-r-idioms/

**Comment by philrosenfield** [comment]
Agreed this is a great idea. I think differentiating these as "modules" or some other term may be useful to separate (if needed) from workshop material. Last week, I gave the git-novice lesson as a 2h session, yeah it could've been 1.5h, guess where the extra 30m was from... The only negative issue I foresee with a tight time constraint is the increased importance of having everyone's install ready to go.

**Comment by ttimbers** [comment]
For these micro-lessons, I strongly support the parsing of our workshop material into smaller digestible chunks. Feedback I have received from new instructors teaching during and after teaching their first workshop is that they already have a difficult time knowing what material to cover in 4 hours (most units take more than 4 hours  to cover in practice), and I think they would be extremely overwhelmed if we asked them to choose how much material to cover in 1 hour. Two other notes on this topic, (1) I think we should focus on the beginning material of each topic as this is what get taught at every workshop, and (2) we should not encourage the micro-lessons to be a faster pace than what we want instructors to teach at workshops. In this training environment, new instructors are building habits which will stick, and thus we want to encourage them to teach as we would want them to teach in an actual workshop.

One question regarding the micro-lesson overview section - in the last point in this section it is suggested that after the micro-lesson the new instructor will "be eligible for badging as a full instructor, at the discretion of the observing instructor." I think it would be good to have some more transparency about what we mean here. Is this a teaching exam? If so we need a rubric for evaluating instructors, and also need to provide the new instructors with evaluation criteria. Or, is this the case that essentially everyone passes unless certain criteria are not met? In this case, what criteria? In either case, spelling this out will help both the new instructor giving the micro-lesson as well as the observing instructor.

Very excited to put these micro-lessons into practice! Great work overall!

**Comment by gvwilson** [comment]
Hi everyone,

If new instructors are going to teach short lessons, I would strongly prefer that these be taken from our existing material, partly so that they become more familiar with it (which is one of the original motivations), and partly so that we're not spreading our (already strained) maintenance effort across a bunch of new stuff.  I'm strongly in favor of breaking our existing lessons up into smaller pieces, and putting more quick assessments at the end of each.

Thanks,
Greg

**Comment by wking** [comment]
On Tue, Mar 24, 2015 at 07:00:51AM -0700, Greg Wilson wrote:
> I'm strongly in favor of breaking our existing lessons up into
> smaller pieces, and putting more quick assessments at the end of
> each.

I agree, but see also some valid pushback from @ChristinaLK in
swcarpentry/shell-extras#3 [1] (with my attempted (postponed) rebuttal
[2] ;).  If we go this route (and I think we should), I think we'll
need to step up our game on streamlining and documenting lesson
maintainance workflows and supporting lesson discoverability.

[1]: https://github.com/swcarpentry/shell-extras/issues/3#issuecomment-83265768
[2]: https://github.com/swcarpentry/shell-extras/issues/3#issuecomment-83921021

**Comment by BillMills** [comment]
Yes and yes to everything @ttimbers and @gvwilson said; amendment to make using the first sections of existing material standard practice inbound.

**Comment by r-gaia-cs** [comment]
@BillMills Thanks for the draft. We will discuss it at our next subcommittee meeting.

**Pull request merged**